cmd/govim/gopls_client.go

```diff
@@ -19,31 +19,53 @@ const (
 var _ protocol.Client = (*govimplugin)(nil)
 
 func (g *govimplugin) ShowMessage(ctxt context.Context, params *protocol.ShowMessageParams) error {
-	g.logGoplsClientf("Asked to show a message: %v", params.Message)
+	g.logGoplsClientf("ShowMessage callback: %v", params.Message)
+	if params.Type != protocol.Error {
+		return nil
+	}
+
+	g.Schedule(func(govim.Govim) error {
+		opts := make(map[string]interface{})
+		opts["mousemoved"] = "any"
+		opts["moved"] = "any"
+		opts["padding"] = []int{0, 1, 0, 1}
+		opts["wrap"] = false
+		opts["close"] = "click"
```

Comment:
Just to confirm: this will place the popup in the middle of the screen, correct?

I think we probably want to be a bit less intrusive than that, add a bit of border and set a suitable highlight colour given the severity of the message. 

So how about the equivalent of:

```vim
call popup_create("we have a problem", {"mousemoved":"any", "moved":"any", "padding":[0,1,0,1], "wrap": "false", "close":"click", "border":[], "highlight": "ErrorMsg", "line":1})
```

That gives something like the following:

<img width="634" alt="Screen Shot 2019-09-09 at 11 16 18" src="https://user-images.githubusercontent.com/3374574/64523058-56673100-d2f3-11e9-9c55-a9b76b70f01b.png">

i.e.

* horizontal centre, place near the top
* error message highlighting
* `=` border

Reply:
@myitcv Thank you so much for a detailed review. Appreciate it. Yes you are right, I have changed my code to reflect what you suggested.

While I was testing the popup, I noticed that the test `show_message.txt` was passing when `vim` was actually throwing an error for using `"wrap": false`. It should be instead `"wrap": "false"`. I just thought it was weird that the test was passing. Please refer the screenshots below.
![Screen Shot 2019-09-09 at 10 02 49 AM](https://user-images.githubusercontent.com/6277978/64537959-c7095000-d2e9-11e9-8b10-afb7b8cb2b67.png)
![Screen Shot 2019-09-09 at 10 02 33 AM](https://user-images.githubusercontent.com/6277978/64537961-c7095000-d2e9-11e9-8668-95ef9ea24f09.png)

Reply:
See comment above.